### ISSUE TITLE
Issue with Database

### ISSUE THREAD

**Issue by lsi117** [lsi117/Notebook#1]
Okay so basically when testing out my app last night I tried to delete a user. When I tried that I got a foreign key error, so I tried to fix that by messing around with my models. After messing around with my models, something wonky started happening to my DB. I figured it would, but I also was able to make my delete users to somewhat work. After messing that up I backed up and made my models back to how they were originally, this still doesn't give me the ability to delete a user because of the foreign key issue. So, I've deleted all of my records using rails c annnnnd then when I did that I ran into another problem! Now, I've read on SO that a table will continue to increment and not leave off wherever you deleted something...so that's throwing nil errors everywhere

I deleted users and posts and comments....following the advice of this link, https://teamtreehouse.com/community/how-do-i-delete-a-user-from-db-in-ruby-on-rails

then after this is when my db keeps incrementing ...so i've read on a couple of SO posts of what to do, but I don't want to proceed since I'm still unsure of how to fix this.
https://stackoverflow.com/questions/15144621/model-delete-all-and-make-the-ids-start-back-at-1

https://stackoverflow.com/questions/5342440/reset-auto-increment-counter-in-postgres

https://stackoverflow.com/questions/7606386/how-to-restart-id-counting-on-a-table-in-postgresql-after-deleting-some-previous

https://stackoverflow.com/questions/41829230/truncate-and-restart-identity-of-a-set-of-tables

Basically, I just want to reset all of my tables ID back to 1 (which will get rid of the NIL errors) and also check the dependencies of my tables....because I need help figuring out why when I try to delete a user it fires off a foreign key error. I think that may have to do with the fact that my users are connected to the comments and posts that they submit, I'm not too sure though.
Thank you

[Code Snippet](https://drive.google.com/open?id=1txhIEp9o0VLnMHFfx5prmOkZkpBKuUax)

[Code Snippet 2](https://drive.google.com/open?id=1-Fv_gPveBVOxMIWjrsMLTnSrvmFXXipY)

IM SO CLOSE TO FINISH AAAAAHHHHH
